eds/interfaces/plugin.py

```diff
@@ -9,14 +9,41 @@ class Plugin():
     schema: Dict = {}
     """json schema for plugin properties"""
 
+    @property
+    def id(self) -> str:
+        """The plugin id.
+
+        Returns:
+            str: The plugin id.
+        """
+        pass
```

Comment:
In the past, for interfaces, I've always set the method body to `raise NotImplementedError()` to make sure that we catch any instances where a method isn't overridden, instead of just letting it silently get called. Thoughts?

Reply:
@jantman added the "NotImplementedError"s